sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -383,6 +352,23 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 	return events
 }
 
+// processSecretForLocalScanning processes and stores secrets to be used for delegated/local scanning.
+func (s *secretDispatcher) processSecretForLocalScanning(secret *v1.Secret, action central.ResourceAction, dockerConfig config.DockerConfig, saName string) {
+	if !env.LocalImageScanningEnabled.BooleanSetting() {
+		// If local scanning is disabled, do not capture secrets.
+		return
+	}
+
+	if action == central.ResourceAction_REMOVE_RESOURCE {
+		if s.regStore.DeleteSecret(secret.GetNamespace(), secret.GetName()) {
```

Comment:
When does this return `false`? When it's `false`, then it looks like we upsert the secret. Should that `return` be moved outside this check?

Reply:
That is a bug, the return should be outside of the condition, will fix and add test.